docs/site/Inside-Loopback-Application.md

```diff
@@ -0,0 +1,161 @@
+
+> [Repositories](https://loopback.io/doc/en/lb4/Repositories.html) in LoopBack represent the `Entity` layer for querying and persisting a domain model. They also connect with other repositories to resolve `Entity relations`.
+
+From the exampple,
+  - To query and persist `User` model data, the example uses a [UserRepository](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/repositories/user.repository.ts).
+  - For the `Order` model, [OrderRepository](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/repositories/order.repository.ts) is defined.
+  - The `UserRepository` uses the entity composition defined in the `User` model to create a has-many relation with the `OrderRepository` 
+
+Now that we have the most of the API implementation next step is to choose appropriate drivers to connect with the back end datasoruces.
+
+> LoopBack has readily available connectors for most [databases](https://loopback.io/doc/en/lb4/Database-connectors.html) and other backend resources like [REST, SOAP, Email, etc](https://loopback.io/doc/en/lb4/Other-connectors.html) to provide easy to use CRUD operations and connection configurations.
+
+In the example, [Redis datasource definition](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/datasources/redis.datasource.json) uses the LoopBack `kv-redis` connector and [MongoDB datasource definition](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/datasources/mongo.datasource.json) uses the LoopBack `mongodb` connector.
+
+## Application startup
+
+The `Shopping` APIs have to be setup with configurations so that they are accessible on a specific port, have a base url, etc. Also we may want to setup [various bootup activities](https://loopback.io/doc/en/lb4/Application.html) if there are tasks to be completed before the APIs are available online.
+
+The [ShoppingApplication class](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/application.ts) is a palette to hold all common configurations and startup activities of the `Shopping` microservice.
+
+The `ShoppingApplication` class extends the `RestApplication` class from the `@loopback\rest` package and so has inherited the capabilities of the in-built [LoopBack Server](https://loopback.io/doc/en/lb4/Server.html) to `boot`, `start` and `stop`.
+The application can now be booted and started by calling the `app.boot` and `app.start` methods respectively. In the example, this call is made from the [index.ts](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/index.ts) file.
```

Comment:
Can we add a summary section? For example, to have a table to map shopping app requirements/features to LB4 facilities/concepts.

Reply:
I have added an API Implementation section in the beginning, that seems to map requirements to LoopBack concepts. Can you check if that fits as a summary ?

Reply:
I have added a summary section at the bottom, slightly different from what you have asked for.